src/fstab-generator/fstab-generator.c

```diff
@@ -692,7 +693,11 @@ static int parse_fstab(bool initrd) {
                                 continue;
                         }
 
-                        if (is_device_path(what)) {
+                        e = getenv_bool_secure("SYSTEMD_FORCE_FSTAB");
```

Comment:
searching for env vars is kinda slow. please move outside of the loop, so it is only done once. or even better, maybe do it lazily the first time we need it. i.e.

```
int force_fstab = -1;

if (force_fstab < 0) {
        r = getenv_bool_secure("SYSTEMD_FORCE_FSTAB");
        if (r < 0 && r != -ENXIO)
                log_debug_errno(r, …");
        force_fstab = e > 0;
}
```

i.e. use -1 as indicator you never parsed this ebfore

Reply:
I've implemented essentially what you had in that snippet, but adjusted for renaming the var per the following review comment